src/borrow_tracker/tree_borrows/tree.rs

```diff
@@ -488,7 +488,12 @@ impl<'tcx> Tree {
 /// Integration with the BorTag garbage collector
 impl Tree {
     pub fn remove_unreachable_tags(&mut self, live_tags: &FxHashSet<BorTag>) {
-        assert!(self.keep_only_needed(self.root, live_tags)); // root can't be removed
+        let root_is_needed = self.keep_only_needed(self.root, live_tags); // root can't be removed
+        assert!(root_is_needed);
+        // Right after the GC runs is a good moment to check if we can
+        // merge some adjacent ranges that were made more similar by
```

Comment:
```suggestion
        // merge some adjacent ranges that were made identical by
```
"similar" is useless, right? Only identical helps?

Reply:
I specifically wanted to avoid "identical" because it does not call the `identical` comparison function. Let's use "equal" instead because it does call `eq`.